src/libs/actions/User.js

```diff
@@ -119,11 +134,48 @@ function setSecondaryLogin(login, password) {
     });
 }
 
+/**
+ * Validates a login given an accountID and validation code
+ *
+ * @param {Number} accountID
+ * @param {String} validateCode
+ */
+function validateLogin(accountID, validateCode) {
+    const isLoggedIn = !_.isEmpty(sessionAuthToken);
+    const redirectRoute = isLoggedIn ? ROUTES.getReportRoute(currentlyViewedReportID) : ROUTES.SIGNIN;
+    Onyx.merge(ONYXKEYS.ACCOUNT, CONST.DEFAULT_ACCOUNT_DATA);
+
+    API.ValidateEmail({
+        accountID,
+        validateCode,
+    }).then((response) => {
+        if (response.jsonCode === 200) {
+            const {email} = response;
+
+            // If the user is unauthenticated, generate a login for them with the returned authtoken and sign then in
+            if (isLoggedIn) {
+                getUserDetails();
+            } else {
+                const success = lodashGet(response, 'message', `Your secondary login ${email} has been validated.`);
+                Onyx.merge(ONYXKEYS.ACCOUNT, {success});
+            }
+        } else {
+            const error = lodashGet(response, 'message', 'Unable to validate login.');
+            Onyx.merge(ONYXKEYS.ACCOUNT, {error});
+        }
+    }).finally(() => {
+        Onyx.merge(ONYXKEYS.ACCOUNT, {loading: false});
+        setCurrentURL(redirectRoute);
```

Comment:
`setCurrentURL()` happens automatically and is just a side effect of navigating

https://github.com/Expensify/Expensify.cash/blob/7300c61f7ccc4399dff7e8d866ae3b52cb6b965d/src/libs/Navigation/NavigationRoot.js#L98-L112